### PR TITLE
[18.09] Not all workflow outputs have a UUID

### PR DESCRIPTION
**Pull request by mvdbeek**
(Majorly!) Broken in https://github.com/galaxyproject/galaxy/pull/7227

### CODE REVIEW

**Comment by nsoranzo** [comment]
Was it failing with `KeyError`?

Can either `data_output['uuid']` or `workflow_output['uuid']` be `''` or `None` ?

**Review by jmchilton — APPROVED** [review]

**Comment by mvdbeek** [comment]
> Can either `data_output['uuid']` or `workflow_output['uuid']` be `''` or `None` ?

Yes, good point!

**Comment by mvdbeek** [comment]
(https://github.com/galaxyproject/galaxy/blob/c228760e1f067111fff0ecf9cf69ed3b019313a2/lib/galaxy/managers/workflows.py#L715)

**Comment by nsoranzo** [comment]
Which means that:
```
data_output.get('uuid', object()) == workflow_output.get('uuid', object())
```
can become:
```
None == None
```
which is `True`.

**Comment by mvdbeek** [comment]
Thanks @nsoranzo, I think the last commit should fix that.

File: lib/galaxy/workflow/modules.py [hunk 1]
```diff
@@ -353,8 +353,11 @@ def get_data_outputs(self):
                     label = workflow_output['label']
                     if not label:
                         label = "%s:%s" % (order_index, workflow_output['output_name'])
+                    workflow_output_uuid = object()
+                    if workflow_output.get('uuid'):
+                        workflow_output_uuid = workflow_output['uuid']
```
Comment (nsoranzo) [inline lib/galaxy/workflow/modules.py]: You can do the same in one line:
```python
workflow_output_uuid = workflow_output.get('uuid') or object()
```
For consistency, I'd also define here `data_output_uuid` in a similar way.

Comment (mvdbeek) [reply lib/galaxy/workflow/modules.py]: Thanks, I've rebase the commit with your suggestions.

File: lib/galaxy/workflow/modules.py [hunk 2]
```diff
@@ -353,8 +353,11 @@ def get_data_outputs(self):
                     label = workflow_output['label']
                     if not label:
                         label = "%s:%s" % (order_index, workflow_output['output_name'])
+                    workflow_output_uuid = object()
```
Comment (nsoranzo) [inline lib/galaxy/workflow/modules.py]: This line is not needed anymore.

Comment (mvdbeek) [reply lib/galaxy/workflow/modules.py]: Of couse 🤦‍♂️ ..., thanks again!

**Review by nsoranzo — APPROVED** [review]

**Pull request merged**